hw_33_online_shop/js/helpers.js

```diff
@@ -0,0 +1,227 @@
+                            </div>
+                            <div class="form-group">
+                            <label for="form-categoryID">Category ID:</label>
+                            ${product ?
+                            `<input type="text" class="form-control" name="categoryID" value="${product.category.id}" required>` :
+                            `<input type="text" class="form-control" name="categoryID" value="" required> `}
+                            </div>
+                            <div class="form-group">
+                            <label for="form-img">Image (URL):</label>
+                            ${product ?
+                            `<input type="text" class="form-control" name="images" value="${product.images[0]}" required>` :
+                            `<input type="text" class="form-control" name="images" value="" required>`}
+                            </div>
+                            <button type="submit" class="button button-hero" id="form-btn">Submit</button>
+                        </form>
+                    </div>
+                </body>
+            </html>`
+        );
+    }, 
+
+    editProducts(product, products, globalContainer, descrContainer) {
+
+        let formWin = window.open("about:blank", "form", "popup");
+        showForm(formWin, product);
+
+        formWin.addEventListener('submit', async (e) => {
+            e.preventDefault();
+            let form = e.target;
+
+            let newProductData = {
+                title: form.title.value,
+                price: form.price.value,
+                description: form.description.value,
+                categoryId: form.categoryID.value,
+                images: [form.images.value]
+            }
+
+            let newProduct;
+
+            if (!product) {
+                newProduct = await createNewProduct(newProductData);
+                products.push(newProduct);
+            } else {
+                let index = products.indexOf(product);
+                newProduct = await updateProduct(product, newProductData);
+                products[index] = newProduct;
+            }
+
+            formWin.close();
+          
+            globalContainer.innerHTML = '';
+            publishProducts(products, globalContainer, descrContainer);
+
+        })
+
+        
+    },
+
+    addCardBtnListeners(itemContainer, product, globalContainer, descrContainer, products) {
+
+        let descrBtn = itemContainer.querySelector('.btn-more');
+        let editBtn = itemContainer.querySelector('.btn-edit');
+
+        descrBtn.addEventListener('click', () => {
+            descrContainer.innerHTML = itemContainer.innerHTML;
+            let cardDetails = descrContainer.querySelector('.card');
+            let descr = descrContainer.querySelector('.descr');
+            makeVisible(descrContainer);
+            makeVisible(descr);
```

Comment:
in that case it's better to refactor `makeVisible` to it be able to work with an array. 

it should work properly for next cases: `makeVisible([descrContainer, descr]) or makeVisible(descrContainer)`